### DISCUSSION TITLE
Migrating history partially

### DISCUSSION THREAD

**Discussion by srohmen** [git-lfs/git-lfs#4616] [Q&A] [answered]
During the course of incrementally migrating a huge repository I stumbled into some performance problems with git lfs migrate. As an example, lets suppose this simplified history on master branch:
A -- B -- C
After migration it is:
A' -- B' -- C'
Now I cherry pick or rebase some more commits on top:
A' -- B' -- C' -- D -- E
Obviously, D and E are not migrated yet and during cherry picking or rebasing the LFS tracking does not apply. When I now execute git lfs migrate again (with the same include statements as before), it will rewrite the complete history again, including the one for A, B and C (although not necessary). This seems not a big deal for small repositories but becomes a huge bottleneck with 100 thousands of LFS tracked files and long histories 

Is there a way to restrict the history rewrite to a range of commits?

**Comment by chrisd8088** [comment, marked as answer]
You might take a look at the `--include-ref` and `--exclude-ref` options; the manual [page](https://github.com/git-lfs/git-lfs/blob/main/docs/man/git-lfs-migrate.1.ronn#include-and-exclude-refs) has some description of those and an example.  In your case, perhaps using `--exclude-ref=<ref to C'>` might work?
